services/src/pro/contexts/postgres.rs

```diff
@@ -173,429 +179,192 @@ where
         Ok(app_ctx)
     }
 
-    async fn schema_version(
-        conn: &PooledConnection<'_, PostgresConnectionManager<Tls>>,
-    ) -> Result<i32> {
-        let stmt = match conn.prepare("SELECT version from version").await {
-            Ok(stmt) => stmt,
-            Err(e) => {
-                if let Some(code) = e.code() {
-                    if *code == SqlState::UNDEFINED_TABLE {
-                        info!("Version table not found. Initializing schema.");
-                        return Ok(0);
-                    }
-                }
-                return Err(error::Error::TokioPostgres { source: e });
-            }
-        };
+    #[allow(clippy::too_many_lines)]
+    /// Creates the database schema. Returns true if the schema was created, false if it already existed.
+    pub(crate) async fn create_pro_schema(
+        conn: PooledConnection<'_, PostgresConnectionManager<Tls>>,
+    ) -> Result<()> {
+        let user_config = get_config_element::<crate::pro::util::config::User>()?;
+
+        conn.batch_execute(
+            &format!(r#"
+            -- TODO: distinguish between roles that are (correspond to) users and roles that are not
+            -- TODO: integrity constraint for roles that correspond to users + DELETE CASCADE
+            CREATE TABLE roles (
+                id UUID PRIMARY KEY,
+                name text UNIQUE NOT NULL
+            );
 
-        let row = conn.query_one(&stmt, &[]).await?;
+            INSERT INTO roles (id, name) VALUES
```

Comment:
Same here, we could put it as .sql files.

Reply:
same problem with the placeholders.

Reply:
cf. idea above

Reply:
7a90b48f1737a64e40a3ab6af6d16deaa12da3ef